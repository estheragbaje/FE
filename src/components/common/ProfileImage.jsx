import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ProfileImage = ({ name, image }) => (
  <StyledImage>
    <div>
      <img src={image} alt="" />
    </div>
    <p>{name}</p>
  </StyledImage>
);

ProfileImage.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

const StyledImage = styled.div`
	width: 100%;
	height: 12rem;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;

	div {
		display: flex;
		justify-content: center;
		align-items: center;

		img {
			width: 50%;
			border-radius: 50%;
		}
	}

	p {
		width: 100%;
		padding: 0.5rem;
		text-align: center;
		display: flex;
		align-items: center;
		justify-content: center;
		font-size: 1.2rem;
	}
`;

export default ProfileImage;
