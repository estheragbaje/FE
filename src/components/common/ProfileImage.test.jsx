import React from 'react';
import { shallow } from 'enzyme';

import ProfileImage from './ProfileImage';

describe('<ProfileImage />', () => {
  const props = {
    name: jest.mock(),
    image: { push: jest.fn() },
  };

  it('renders the ProfileImage component correctly', () => {
    const wrapper = shallow(<ProfileImage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
