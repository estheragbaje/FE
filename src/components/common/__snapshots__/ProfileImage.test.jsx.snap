// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileImage /> renders the ProfileImage component correctly 1`] = `
<styled.div>
  <div>
    <img
      alt=""
      src={
        Object {
          "push": [MockFunction],
        }
      }
    />
  </div>
  <p>
    <Component />
  </p>
</styled.div>
`;
